check_manifest.py

```diff
@@ -961,10 +961,15 @@ def check_manifest(source_tree='.', create=False, update=False,
             ui.info_continue(": %d files and directories" % len(sdist_files))
             version = extract_version_from_filename(sdist_filename)
         existing_source_files = list(filter(os.path.exists, all_source_files))
-        missing_source_files = sorted(set(all_source_files) - set(existing_source_files))
+        existing_source_files_with_broken_simlinks = list(filter(os.path.lexists, all_source_files))
+        missing_source_files = sorted(set(all_source_files) - set(existing_source_files_with_broken_simlinks))
         if missing_source_files:
             ui.warning("some files listed as being under source control are missing:\n%s"
                        % format_list(missing_source_files))
+        broken_symlinks = set(existing_source_files_with_broken_simlinks) - set(existing_source_files)
+        if broken_symlinks:
+            ui.warning("some symbolic links listed as being under source control are pointing to missing files:\n%s"
+                       % format_list(["%s -> %s (missing)" % (x, os.path.realpath(x)) for x in broken_symlinks]))
```

Comment:
Hmm, is it better to use os.path.realpath(x) here, or is it better to use os.readlink(x) here?

Reply:
I've switch to `os.readlink` to try it but the result ended up being the same?

Apparently it can returns relative path sometime according to the documentation but I don't know when. I guess this will be more readable in some situations?

Reply:
I suppose the difference would be if you had two symlinks, like a -> b and b -> c.  os.realpath('a') would return 'c', but os.readlink('a') would return 'b'.

But we know that in this case b doesn't exist, so it can't be a symlink to c.

How about symlinks to directories?  a -> b/c, where b -> d, and d/c is missing.  os.realpath('a') is 'd/c', and os.readlink('a') is 'b/c'.  Which would be better to show in the error message?

To me readlink() feels a bit better.